libsolidity/ast/Types.cpp

```diff
@@ -2607,6 +2607,41 @@ bool FunctionType::isExplicitlyConvertibleTo(Type const& _convertTo) const
 	return _convertTo.category() == category();
 }
 
+bool FunctionType::isImplicitlyConvertibleTo(Type const& _convertTo) const
+{
+	if (_convertTo.category() != category())
+		return false;
+	
+	FunctionType const& convertTo = dynamic_cast<FunctionType const&>(_convertTo);
+	
+	if (
+		m_kind != convertTo.m_kind ||
+		m_parameterTypes.size() != convertTo.m_parameterTypes.size() ||
+		m_returnParameterTypes.size() != convertTo.m_returnParameterTypes.size()
+	)
+		return false;
+	
+	auto typeCompare = [](TypePointer const& _a, TypePointer const& _b) -> bool { return *_a == *_b; };
+	if (
+		!equal(m_parameterTypes.cbegin(), m_parameterTypes.cend(), convertTo.m_parameterTypes.cbegin(), typeCompare) ||
```

Comment:
There is a function `hasEqualArgumentTypes` - you could add another called `hasEqualArgumentAndReturnTypes` and call it here. Actually in that function it should be `Parameters`, not `Arguments` - could you also fix that at the same time, please?

Reply:
Ah now I see that most of this is taken from `operator==` - could you reduce code duplication somehow?

Reply:
I've reduced code duplication by implementing operator== in terms of isImplicitlyConvertible